Mage/src/main/java/mage/target/common/TargetAnyTargetAmount.java

```diff
@@ -14,25 +14,20 @@ public class TargetAnyTargetAmount extends TargetPermanentOrPlayerAmount {
     private static final FilterPermanentOrPlayer defaultFilter
             = new FilterAnyTarget("targets");
 
-    public TargetAnyTargetAmount(int amount) {
-        this(amount, 0);
+    public TargetAnyTargetAmount(int amount, int minNumberOfTargets) {
+        this(amount, minNumberOfTargets, amount);
     }
 
-    public TargetAnyTargetAmount(int amount, int maxNumberOfTargets) {
-        // 107.1c If a rule or ability instructs a player to choose “any number,” that player may choose
-        // any positive number or zero, unless something (such as damage or counters) is being divided
-        // or distributed among “any number” of players and/or objects. In that case, a nonzero number
-        // of players and/or objects must be chosen if possible.
-        this(StaticValue.get(amount), maxNumberOfTargets);
-        this.minNumberOfTargets = 1;
+    public TargetAnyTargetAmount(int amount, int minNumberOfTargets, int maxNumberOfTargets) {
+        this(StaticValue.get(amount), minNumberOfTargets, maxNumberOfTargets);
     }
 
     public TargetAnyTargetAmount(DynamicValue amount) {
-        this(amount, 0);
+        this(amount, 0, 0);
```

Comment:
Can you please explain the usage of maxNumberOfTargets = 0 ? I would have expected Integer.MAX_VALUE, and I can't figure out where in the code 0 is getting handled.

Reply:
Potentially related: #9727. There's some weirdness going on with 0 being used as an "unlimited choices" in targeting. Has some undesirable outcomes in cases of playing cards like [Sporocyst](https://scryfall.com/card/40k/98/sporocyst) with X=0.

Reply:
Yeah, I think I saw a comment somewhere that led me down that path. Looking over it again, I couldn't find any problem with using Integer.MAX_VALUE instead, so I've switched to that.

Reply:
Thanks for the info - that should probably be broken out as its own issue, and scope of fixing it properly researched. But at least this PR won't add to it anymore.